src/ngp_algorithms/TurbViscKsgsAlg.C

```diff
@@ -48,11 +48,16 @@ TurbViscKsgsAlg::execute()
   const auto& meshInfo = realm_.mesh_info();
   const auto ngpMesh = meshInfo.ngp_mesh();
   const auto& fieldMgr = meshInfo.ngp_field_manager();
-  const auto tke = fieldMgr.get_field<double>(tke_);
-  const auto density = fieldMgr.get_field<double>(density_);
-  const auto dualNodalVolume = fieldMgr.get_field<double>(dualNodalVolume_);
+  auto tke = fieldMgr.get_field<double>(tke_);
+  auto density = fieldMgr.get_field<double>(density_);
+  auto dualNodalVolume = fieldMgr.get_field<double>(dualNodalVolume_);
   auto tvisc = fieldMgr.get_field<double>(tvisc_);
 
+  tke.sync_to_device();
```

Comment:
Question regarding API, `NgpFieldManager::get_field` is calling `stk::mesh::get_updated_ngp_field`. the name seems to indicate that it will have updated the field on device, why are we syncing to device here?

Reply:
We're regretting the name 'get_updated_ngp_field'. In current semantics it means the field is updated structurally, compatible with mesh structure, but doesn't mean data is sync'd. STK needs to figure out a better way to clarify that separation of concerns.